rust/src/importer.rs

```diff
@@ -39,12 +42,68 @@ pub fn tweak_imported_file_info(
     }
 }
 
+/// Apply filtering and manipulation logic to an RPM file before importing.
+///
+/// This returns whether the entry should be ignored by the importer.
+pub fn importer_compose_filter(
+    path: &str,
+    mut file_info: Pin<&mut crate::FFIGFileInfo>,
+    skip_extraneous: bool,
+) -> CxxResult<bool> {
+    let mut file_info = file_info.gobj_wrap();
+    match import_filter(path, &mut file_info, skip_extraneous)? {
+        RepoCommitFilterResult::Allow => Ok(false),
+        RepoCommitFilterResult::Skip => Ok(true),
+        x => unreachable!("unknown commit result '{}' for path '{}'", x, path),
+    }
+}
+
+#[context("Analyzing {}", path)]
+fn import_filter(
+    path: &str,
+    file_info: &mut gio::FileInfo,
+    skip_extraneous: bool,
+) -> Result<RepoCommitFilterResult> {
+    // Sanity check that RPM isn't using CPIO id fields.
+    {
+        let uid = file_info.get_attribute_uint32("unix::uid");
```

Comment:
I do this "introduce a new scope" thing when the return value is expensive - e.g. we're holding onto a file descriptor or large `malloc()` buffer.  But here `uid/gid` are just integers.  Any reason for the new scope?